File: tests/test-terraria/src/main/java/pize/tests/terraria/tile/TileType.java
package pize.tests.terraria.tile;

public enum TileType{

    AIR(0, false),
    DIRT(1, true),
    STONE(2, true),
    GRASS(3, true);

    public final short id;
    public final boolean collidable;

    TileType(int id, boolean collidable){
        this.id = (short) id;
        this.collidable = collidable;
    }

}

File: tests/test-terraria/src/main/kotlin/pize/tests/terraria/tile/TileType.kt
package pize.tests.terraria.tile

enum class TileType(id: Int, val collidable: Boolean) {
    AIR(0, false),
    DIRT(1, true),
    STONE(2, true),
    GRASS(3, true);

    val id: Short

    init {
        this.id = id.toShort()
    }
}
